src/libaktualizr/bootloader/bootloader.cc

```diff
@@ -169,13 +174,26 @@ void Bootloader::rebootFlagClear() {
 }
 
 void Bootloader::reboot(bool fake_reboot) {
+  size_t len = reboot_command_.length();
+  struct stat st;
+  char *cmd = NULL, full_cmd[256];
+
   if (fake_reboot) {
     boost::filesystem::remove(reboot_sentinel_);
   } else {
-    sync();
     if (setuid(0) != 0) {
       LOG_ERROR << "Failed to set/verify a root user so cannot reboot system programmatically";
+      return;
+    }
+
+    reboot_command_.copy(full_cmd, len);
```

Comment:
Seems odd (and unsafe) to copy in a fixed buffer with the length of the original string as a parameter. Doing `.copy(full_cmd, sizeof(full_cmd)-1)` would be more logical.

However, even then, some points still bother me, as I think this part could be much simpler:
- handling c strings when the c++ standard library should be sufficient
- do we really need to parse the command and stat the file? Just `system()` and checking the return code + logging errors should be plenty
- we'd probably want to run `sync()` in any case, even with a custom command
- maybe not defaulting to `::reboot` if the default command fails. It's really not the nominal case (probably misconfiguration) and rebooting anyway could make debugging more difficult